src/app/models/form.server.model.ts

```diff
@@ -313,6 +313,9 @@ const compileFormModel = (db: Mongoose): IFormModel => {
           'msgSrvcName must be alphanumeric, dashes and slashes are allowed',
         ],
       },
+
+      hasSubmissionLimit: { type: Boolean, default: false },
+      submissionLimit: { type: Number, default: 1000 },
```

Comment:
Apologies if this is coming a bit late, but do you think the implementation could avoid storing the `hasSubmissionLimit` boolean, and switch to only storing `submissionLimit` which could be a `Number` that defaults to `undefined?

 I'm suggesting this because the `hasSubmissionLimit` value appears to be necessary only because of UI requirements, but it can also be inferred from `submissionLimit` upon initialisation of the frontend controller.

Reply:
More ideally, related controls would also be grouped and nested together in the schema, but we have this nasty bug where nested keys overwrite the entire object...

Reply:
I think in general the logic is simpler with the `hasSubmissionLimit` than without on the admin page, and there is the ability to preserve what the user has entered in the `submissionLimit` field between toggling the `hasSubmissionLimit` field on and off.

That said, if you think having one field is better, I can look into changing it.

Reply:
Perhaps we can discuss this on Tues given the UX implications. I am still leaning towards having it as a single field so that there's no chance of an inconsistent database state.